obspy/signal/spectral_estimation.py

```diff
@@ -1918,24 +1917,23 @@ def plot(self, filename=None, show_coverage=True, show_histogram=True,
                 ax.plot(xdata, noise_model, '0.4', linewidth=2, zorder=10)
         
         if show_earthquakes:
-            for eqk in earthquakes:
+            for (magnitude, distance), (periods, power) in earthquakes.items():
                 # Eq.1 Cauzzi & Clinton (2013)
-                ydata=[v/((1./earthquakes[eqk][0][iv])**(-.5)) for iv,v in enumerate(earthquakes[eqk][1])]
-                ydata=[20*np.log10(v/2) for v in ydata]
-                if (eqlims[0] < eqk[0] and eqk[0] < eqlims[1] and
-                    eqlims[2] < eqk[1] and eqk[1] < eqlims[3] and
+                ydata=[v / ((1. / periods[iv]) ** (-.5)) for iv, v in enumerate(power)]
+                ydata=[20 * np.log10(v / 2) for v in ydata]
+                if (eqlims[0] < magnitude and magnitude] < eqlims[1] and
+                    eqlims[2] < distance and distance < eqlims[3] and
                     min(ydata) < self.db_bin_edges[-1]):
                     if xaxis_frequency:
-                        xdata =  earthquakes[eqk][0]
+                        xdata =  periods
                     else:
-                        xdata = [1.0/v for v in earthquakes[eqk][0]]
-                    ax.plot(xdata, ydata,
-                            '0.4', linewidth=2)                
+                        xdata = [1.0/v for v in periods]
```

Comment:
`1.0 / periods` or maybe even define frequencies once at top, since it's used multiple times

```python
power = np.array(power)
periods = np.array(periods)
frequencies = 1.0 / periods
....
xdata = frequencies
```

Reply:
OK.